### PR TITLE
feat!: Run plugin on live deploy URL by default 

### LINKED ISSUE #587: Run the plugin onSuccess against live deploy by default

### PR DESCRIPTION
**Pull request by aitchiss**
Closes #587 

We previous released the ability to run the plugin during the `onSuccess` build step as experimental/undocumented functionality. We've had good feedback from users and are now ready to roll this out as the default behaviour.

**⚠️ This PR has breaking changes**

The plugin will now run _after_ a deploy has completed auditing the live deploy rather than the statically served build folder. This will:

- Allow the plugin to be used on ISR/SSR sites
- Speed up "time to deploy" since the plugin run no longer holds up the deploy
- Allow the plugin to run in an environment truer to production

The main side effect of this change is that if score thresholds have been configured, **the plugin will no longer fail the deploy if those scores are not met**. 

To continue running the plugin `onPostBuild` (i.e. before the deploy is complete, using the statically served build folder) and maintain the "fail on thresholds" functionality, set `fail_deploy_on_score_thresholds` to `'true'` in `netlify.toml`:

```toml
[[plugins]]
  package = "@netlify/plugin-lighthouse"

  # Set the plugin to run prior to deploy, failing the build if minimum thresholds aren't set 
  [plugins.inputs]
    fail_deploy_on_score_thresholds = "true"

  # Set minimum thresholds for any report area
  [plugins.inputs.thresholds]
    performance = 0.9
```

If your configuration currently relies on `serveDir` to ensure the correct page(s) are audited, please note this will have no impact on the new default plugin behaviour. Either continue to run during the `onPostBuild` stage via the above steps, or amend your `path` input(s) to ensure the full page route is specified.

For example -

Old configuration relying on `serveDir`:

```toml
[[plugins]]
  package = "@netlify/plugin-lighthouse"

  # Serves the 'pages' directory and audits the file about.html
  # On the live deploy this may correspond to http://1234--yoursite.netlify.app/pages/about
  [[plugins.inputs.audits]]
    serveDir = "pages"
    path = "about.html"
```

New configuration not relying on `serveDir`:

```toml
[[plugins]]
  package = "@netlify/plugin-lighthouse"

  [[plugins.inputs.audits]]
    path = "pages/about"
```

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -6,12 +6,11 @@ import getUtils from './lib/get-utils/index.js';
...
 
 export default function lighthousePlugin(inputs) {
-  // Run onPostBuild by default, unless LIGHTHOUSE_RUN_ON_SUCCESS env var is set to true, or run_on_success is specified in plugin inputs
+  // Run onSuccess by default, unless inputs specify we should fail_deploy_on_score_thresholds
   const defaultEvent =
-    inputs?.run_on_success === 'true' ||
-    process.env.LIGHTHOUSE_RUN_ON_SUCCESS === 'true'
-      ? 'onSuccess'
-      : 'onPostBuild';
+    inputs?.fail_deploy_on_score_thresholds === 'true'
```
Comment (aitchiss) [inline src/index.js:11]: This is the main change in this PR - pretty much everything else is docs changes or updating tests to include `fail_deploy_on_score_thresholds` / remove the old env variable for onSuccess

**Comment by aitchiss** [comment]
## Test steps for this PR

### Check the new default behaviour

Using a test site,
- Install this branch version of the plugin
- Make sure you don't have the `fail_deploy_on_score_thresholds` input set
- Deploy your test site
- Lighthouse output should appear during the `onSuccess` step
- Lighthouse reports should appear in the UI
- Viewing the full lighthouse report should show the deploy URL in the header (not `localhost`)
- Check setting thresholds that your deploy won't achieve. The deploy should complete as normal, but the plugin runs in the deploy summary should note the failing threshold
- Check you can still audit multiple pages by specifying them in the `netlify.toml` and they all run `onSuccess`

### Check the old `onPostBuild` behaviour

Using a test site
- Install this branch version of the plugin
- Set the plugin input `fail_deploy_on_score_thresholds = "true"` as per the instructions above
- Deploy your test site
- Lighthouse output should appear during the `onPostBuild` step (and the deploy should take longer while this runs)
- Lighthouse reports should appear in the UI
- Viewing the full lighthouse report should show `localhost` in the header (not the deploy URL)
- Check setting thresholds that your deploy won't achieve. The deploy should fail as a result of the thresholds
- Check you can still audit multiple pages by specifying them in the `netlify.toml` and they all run `onPostBuild`

File: README.md [hunk 2]
```diff
@@ -173,21 +184,13 @@ yarn local
 
 ## Preview Lighthouse results within the Netlify UI
 
-Netlify offers an experimental feature through Netlify Labs that allows you to view Lighthouse scores for each of your builds on your site's Deploy Details page with a much richer format.
```
Comment (aitchiss) [inline README.md:176]: This section was incorrectly referencing the feature being in Labs, seems it got missed in a previous cleanup. I've also added some new screenshots to bring them a bit more up to date

File: package-lock.json [hunk 3]
```diff
@@ -6,7 +6,7 @@
   "packages": {
     "": {
       "name": "@netlify/plugin-lighthouse",
-      "version": "4.0.7",
+      "version": "4.1.1",
```
Comment (aitchiss) [inline package-lock.json:9]: Missed by previous release(s)

**Review by benedfit — DISMISSED** [review]

**Comment by aitchiss** [comment]
All manual tests/edge cases checked, just putting a pin in this for now so docs has a chance to weigh in 📌

**Review by jackbrewer — DISMISSED** [review]
LGTM! Thank you so much for getting this over the line 🚀

**Review by benedfit — APPROVED** [review]

**Review by klavavej — DISMISSED** [review]
The changes you made to the README look good to me, but I found a few pre-existing things in code blocks that could use further tidying up to align with the other changes you made.

**Comment by aitchiss** [comment]
> The changes you made to the README look good to me, but I found a few pre-existing things in code blocks that could use further tidying up to align with the other changes you made.

Great call outs - thank you! I've updated those sections now

**Review by klavavej — APPROVED** [review]
Thanks for the changes - LGTM!

**Comment by dsolanorush** [comment]
Great work @aitchiss!
